### DISCUSSION TITLE
I need GraphQL APIs for GitHub, mind if I add it here?

### DISCUSSION THREAD

**Discussion by manchicken** [XAMPPRocky/octocrab#263] [Ideas]
I need to be able to use GraphQL APIs to do lookups for SAML user IDs. Do you mind if I add that here?

I don't see many great GraphQL crates, I'm debating just using something that knows how to parse and validate GraphQL, and then doing the transport myself using the `http` crate we discussed before rather than `reqwest`.

Thoughts?

(This is time-sensitive for me)

**Comment by manchicken** [comment]
I'm thinking of using this module: https://github.com/graphql-rust/graphql-client

It's CI is failing (mostly appears to be clippy and prettier that's bombing, not actual tests), but it is the closest I can find to useful.

Thoughts?

**Comment by manchicken** [comment]
Grumble... getting away from `reqwest` will be harder with this module, it has its own transport layer that makes things super easy-to-use, but it's based on `reqwest`.

**Comment by XAMPPRocky** [comment]
Thank you for your issue! I'm not sure I fully understand the feature request. Octocrab should already integrate with graphql through the `graphql` function. You can use whatever graphql library you like and it should work.

**Comment by manchicken** [comment]
Ack! Somehow I didn't notice the `graphql` function there, my bad. Thanks!

**Comment by manchicken** [comment]
You can close this discussion, the `graphql` function 100% solved my problem.
